### PR TITLE
kselftest.sh: remove skipped

### PR DESCRIPTION
**Pull request by alexandrasp**
Remove perl script responsible to write skipped tests on test result output file.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

### CODE REVIEW

File: automated/linux/kselftest/kselftest.sh [hunk 1]
```diff
@@ -204,8 +204,6 @@ echo "skiplist:"
 echo "========================================"
 while read -r skip_regex; do
     echo "$skip_regex"
-    # Remove matching tests from list of tests to run and report it as skipped
-    perl -i -ne 'if (s|^('"${skip_regex}"')$|\1 skip|) { print STDERR; } else { print; }' kselftest-list.txt 2>>"${RESULT_FILE}"
 done < "${skips}"
```
Comment (gctucker) [inline automated/linux/kselftest/kselftest.sh]: Maybe remove the whole `while` block instead of just the `perl` line?

**Review by gctucker — COMMENTED** [review]
Tested OK on staging, here's a production set of results with `kselftest-lib` showing things like `netfilter` and `ftrace`:
https://linux.kernelci.org/test/plan/id/60f598f90ceddb52a7116113/

and here's a corresponding set of results on staging without the unwanted ones (i.e. only the 6 passing test cases):
https://staging.kernelci.org/test/plan/id/60f5af021e8c925c06c28cb1/

**Comment by alexandrasp** [comment]
> Tested OK on staging, here's a production set of results with `kselftest-lib` showing things like `netfilter` and `ftrace`:
> https://linux.kernelci.org/test/plan/id/60f598f90ceddb52a7116113/

thanks for commenting @gctucker

**Comment by gctucker** [comment]
@alexandrasp Have you tried with a kselftest collection that actually had test cases in a skiplist, to check that's still working?

**Comment by alexandrasp** [comment]
> @alexandrasp Have you tried with a kselftest collection that actually had test cases in a skiplist, to check that's still working?

Yes, I have tried. In this case we won't have any results, is this what we want?

**Comment by gctucker** [comment]
Well, no that's not what we want.  If there is a skiplist and a test case is actually skipped, it should be reported as skipped.  So removing the block about skipped tests is not solution, it should be fixed instead.

**Comment by alexandrasp** [comment]
> Well, no that's not what we want. If there is a skiplist and a test case is actually skipped, it should be reported as skipped. So removing the block about skipped tests is not solution, it should be fixed instead.

@gctucker take a look at this results: https://staging.kernelci.org/test/plan/id/60f7e0b1932e6ee370c28d07/
I am experimenting more, the same thing happened with a collection with tests in skiplist (netfilter:bridge_brouter.sh):
https://lava.collabora.co.uk/scheduler/job/4226103#L3284

**Comment by gctucker** [comment]
It looks like what is needed to verify this is now working as expected would be to pick 2 or 3 collections, and for each of them list the test cases defined in the kernel tree, the ones in the skiplist, and the results from the test job.

All the tests defined in the kernel tree should appear in the results.  Those on the skiplist should not be run and their result should be "skipped".  All the others should be either "pass" or "fail".

**Comment by alexandrasp** [comment]
@gctucker @Lakshmipathi was that last force-pushed right?

**Comment by gctucker** [comment]
@alexandrasp Yes, please rebase.

**Comment by alexandrasp** [comment]
collection without test cases in skiplist and all tests pass:

https://staging.kernelci.org/test/plan/id/60f68b2f6bedc2aa6bc28ca2/
https://staging.kernelci.org/test/plan/id/60f68b14550aa5844dc28d0a/

collection without test cases in skiplist and with test cases skipped:

https://staging.kernelci.org/test/plan/id/60f68b09550aa5844dc28cab/
https://staging.kernelci.org/test/plan/id/60f68af414c7858406c28cba/

**Comment by alexandrasp** [comment]
It seems that there is still a problem when it tries to run tests that are in skiplist, I am investigating what is happening.

**Pull request closed without merging**